docs/reference-manual/native-image/assets/reachability-metadata-schema-v1.0.0.json

```diff
@@ -0,0 +1,362 @@
+            "type": "boolean",
+            "default": false
+          }
+        },
+        "additionalProperties": false
+      }
+    },
+    "jni": {
+      "type": "array",
+      "default": [],
+      "items": {
+        "title": "Elements that should be registered for JNI for a specified type",
+        "type": "object",
+        "properties": {
+          "reason": {
+            "title": "Reason for the element's inclusion",
+            "$ref": "#/$defs/reason"
+          },
+          "condition": {
+            "title": "Condition under which the class should be registered for JNI",
+            "$ref": "#/$defs/condition"
+          },
+          "type": {
+            "title": "Type descriptor of the class that should be registered for JNI",
+            "$ref": "#/$defs/type"
+          },
+          "methods": {
+            "title": "List of methods that should be registered for the type declared in <type>",
+            "type": "array",
+            "default": [],
+            "items": {
+              "title": "Method descriptor of the method that should be registered for JNI",
+              "$ref": "#/$defs/method"
+            }
+          },
+          "fields": {
+            "title": "List of class fields that can be read or written to for the type declared in <type>",
+            "type": "array",
+            "default": [],
+            "items": {
+              "title": "Field descriptor of the field that should be registered for JNI",
+              "$ref": "#/$defs/field"
+            }
+          },
+          "allDeclaredMethods": {
+            "title": "Register all declared methods from the type for JNI access",
+            "type": "boolean",
+            "default": false
+          },
+          "allDeclaredFields": {
+            "title": "Register all declared fields from the type for JNI access",
+            "type": "boolean",
+            "default": false
+          },
+          "allDeclaredConstructors": {
+            "title": "Register all declared constructors from the type for JNI access",
+            "type": "boolean",
+            "default": false
+          },
+          "allPublicMethods": {
+            "title": "Register all public methods from the type for JNI access",
+            "type": "boolean",
+            "default": false
+          },
+          "allPublicFields": {
+            "title": "Register all public fields from the type for JNI access",
+            "type": "boolean",
+            "default": false
+          },
+          "allPublicConstructors": {
+            "title": "Register all public constructors from the type for JNI access",
+            "type": "boolean",
+            "default": false
+          }
+        },
+        "additionalProperties": false
+      }
+    },
+    "reason": {
+      "type": "string",
+      "default": []
```

Comment:
This is defined as type `string` but has a default value of an `array`.

What do you think about using:
```
      "oneOf": [
        {
          "type": "string"
        },
        {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      ],
      "default": ""
```

like in `comment`?

Reply:
@zakkak we will implement your proposal in the follow-up PR. Your proposal seems more accurate.